dotnet pack not working with the net5.0-windows10.0.19041.0 TFM

### Issue Description

When I run `dotnet pack` against `net6.0-windows10.0.19041.0` or `net5.0-windows10.0.19041.0` I get the following exception

```
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2505,6): error MSB4062: The "Microsoft.Build.Tasks.FindInvalidProjectReferences" task could not be loaded from the assembly Microsoft.Build.Tasks.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a.  Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\Users\shane\Source\Repos\ANugetTest\ClassLibrary1\ClassLibrary1.csproj]
```

### Steps to Reproduce
[ANugetTest.zip](https://github.com/dotnet/msbuild/files/6172057/ANugetTest.zip)

I've included a sln here that has a net6 and a net5 target.

```cmd
C:\ANugetTest\Net6Target>dotnet pack
```

### Expected Behavior
A nuget package is created

### Actual Behavior

The following exception is thrown

```
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2505,6): error MSB4062: The "Microsoft.Build.Tasks.FindInvalidProjectReferences" task could not be loaded from the assembly Microsoft.Build.Tasks.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a.  Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\Users\shane\Source\Repos\ANugetTest\ClassLibrary1\ClassLibrary1.csproj]
```

### Analysis
I tried setting
`<FindInvalidProjectReferences>false</FindInvalidProjectReferences>`

But it didn't help

`msbuild /t:pack` works

Some additional context
https://microsoft.visualstudio.com/OS/_workitems/edit/32211926

It looks like 
`dotnet build` doesn't work and fails with the same exception

### Versions & Configurations
`msbuild -version`
16.10.0.11801

`dotnet --version`
6.0.100-preview.2.21155.3

### Attach a binlog

[msbuild (2).zip](https://github.com/dotnet/msbuild/files/6172101/msbuild.2.zip)

First pass investigation notes:

Can repro this bug. Doing a little bit of digging:

The FindInvalidProjectReferences task is imported like so in `Microsoft.Common.tasks`:
```
<UsingTask TaskName="Microsoft.Build.Tasks.FindInvalidProjectReferences"          AssemblyName="Microsoft.Build.Tasks.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" Condition="'$(MSBuildAssemblyVersion)' != ''" />
```

In the `Microsoft.Build.Tasks.Core` assembly under my installed `C:\Program Files\dotnet\sdk\6.0.100-preview.2.21155.3` folder, I don't see `FindInvalidProjectReferences`.

![image](https://user-images.githubusercontent.com/4691428/112349866-538a2200-8c86-11eb-97a7-7a589a681065.png)


on `16.10.0.12417` msbuild and running `msbuild /t:pack` I get this error:
```
"C:\Users\bevillal\Downloads\ANugetTest\Net6Target\Net6Target.csproj" (pack target) (1) ->
(_GetSdkToolsPathsFromSdk target) ->
  C:\Users\bevillal\.nuget\packages\microsoft.projectreunion.foundation\0.5.0-prerelease\build\Microsoft.ApplicationMod
el.Resources.PriGen.targets(833,5): error MSB4062: The "Microsoft.Build.AppxPackage.GetSdkFileFullPath" task could not
be loaded from the assembly C:\Program Files (x86)\Microsoft Visual Studio\2019\Dogfood\MSBuild\Microsoft\VisualStudio\
v16.0\AppxPackage\Microsoft.Build.AppxPackage.dll. Could not load file or assembly 'file:///C:\Program Files (x86)\Micr
osoft Visual Studio\2019\Dogfood\MSBuild\Microsoft\VisualStudio\v16.0\AppxPackage\Microsoft.Build.AppxPackage.dll' or o
ne of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct,
 that the assembly and all its dependencies are available, and that the task contains a public class that implements Mi
crosoft.Build.Framework.ITask. [C:\Users\bevillal\Downloads\ANugetTest\Net6Target\Net6Target.csproj]
```

--- a/ref/Microsoft.Build.Tasks.Core/netstandard/Microsoft.Build.Tasks.Core.cs
+++ b/ref/Microsoft.Build.Tasks.Core/netstandard/Microsoft.Build.Tasks.Core.cs
@@ -290,6 +290,18 @@ namespace Microsoft.Build.Tasks
         public bool MatchFileNameOnly { get { throw null; } set { } }
         public override bool Execute() { throw null; }
     }
+    public partial class FindInvalidProjectReferences : Microsoft.Build.Tasks.TaskExtension
+    {
+        public FindInvalidProjectReferences() { }
+        [Microsoft.Build.Framework.OutputAttribute]
+        public Microsoft.Build.Framework.ITaskItem[] InvalidReferences { get { throw null; } }
+        public Microsoft.Build.Framework.ITaskItem[] ProjectReferences { get { throw null; } set { } }
+        [Microsoft.Build.Framework.RequiredAttribute]
+        public string TargetPlatformIdentifier { get { throw null; } set { } }
+        [Microsoft.Build.Framework.RequiredAttribute]
+        public string TargetPlatformVersion { get { throw null; } set { } }
+        public override bool Execute() { throw null; }
+    }
     public partial class FindUnderPath : Microsoft.Build.Tasks.TaskExtension
     {
         public FindUnderPath() { }

--- a/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj
+++ b/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj
@@ -99,7 +99,6 @@
     <Compile Remove="CSharpTokenizer_Tests.cs" />
     <Compile Remove="DependentAssembly_Tests.cs" />
     <Compile Remove="ErrorWarningMessage_Tests.cs" />
-    <Compile Remove="FindInvalidProjectReferences_Tests.cs" />
     <Compile Remove="ResourceHandling\GenerateResourceOutOfProc_Tests.cs" />
     <Compile Remove="ResourceHandling\ResGen_Tests.cs" />
     <Compile Remove="ResourceHandling\ResGenDependencies_Tests.cs" />

--- a/src/Tasks/Microsoft.Build.Tasks.csproj
+++ b/src/Tasks/Microsoft.Build.Tasks.csproj
@@ -387,6 +387,7 @@
     <Compile Include="FindAppConfigFile.cs">
       <ExcludeFromStyleCop>true</ExcludeFromStyleCop>
     </Compile>
+    <Compile Include="FindInvalidProjectReferences.cs" />
     <Compile Include="GetFrameworkPath.cs">
       <ExcludeFromStyleCop>true</ExcludeFromStyleCop>
     </Compile>
@@ -604,7 +605,6 @@
     <Compile Include="ComReferenceWrapperInfo.cs">
       <ExcludeFromStyleCop>true</ExcludeFromStyleCop>
     </Compile>
-    <Compile Include="FindInvalidProjectReferences.cs" />
     <Compile Include="GenerateBootstrapper.cs">
       <ExcludeFromStyleCop>true</ExcludeFromStyleCop>
     </Compile>

